From ed763b88b2523f3b2920653756fcf05fef36f0c7 Mon Sep 17 00:00:00 2001
From: Ghennadi Procopciuc <Ghennadi.Procopciuc@nxp.com>
Date: Fri, 8 Jan 2021 07:51:30 +0200
Subject: [PATCH 33/73] gpio: s32-gen1: Don't clear IRQ type when masking a
 GPIO IRQ

The type of the interrupt (raising / falling) should not be cleared
during mask callback. This is visible during resume operation when the
driver fails to restore the trigger edge of the EIRQs.

Issue: ALB-6099
Signed-off-by: Ghennadi Procopciuc <Ghennadi.Procopciuc@nxp.com>
---
 drivers/gpio/gpio-siul2-s32gen1.c | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/drivers/gpio/gpio-siul2-s32gen1.c b/drivers/gpio/gpio-siul2-s32gen1.c
index ba2befeaa63e..15c60802fedb 100644
--- a/drivers/gpio/gpio-siul2-s32gen1.c
+++ b/drivers/gpio/gpio-siul2-s32gen1.c
@@ -439,15 +439,10 @@ static void siul2_gpio_irq_mask(struct irq_data *data)
 	unsigned long flags;
 	u32 direr0_val;
 	u32 disr0_val;
-	int err;
 
 	if (!siul2_is_valid_eirq(gpio_dev, eirq))
 		return;
 
-	err = siul2_gpio_irq_set_type(data, IRQ_TYPE_NONE);
-	if (err)
-		return;
-
 	spin_lock_irqsave(&gpio_dev->lock, flags);
 
 	regmap_read(gpio_dev->irqmap, SIUL2_DIRER0, &direr0_val);
-- 
2.25.1

